python-backend/app/api/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
+
+import uuid
+
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import validates
+from sqlalchemy.schema import FetchedValue
+from app.extensions import db
+
+from . import permit_amendment_status_code, permit_amendment_type_code
+from ....utils.models_mixins import AuditMixin, Base
+
+
+class PermitAmendment(AuditMixin, Base):
+    __tablename__ = 'permit_amendment'
+    permit_amendment_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
+    permit_amendment_guid = db.Column(UUID(as_uuid=True))
+    permit_id = db.Column(db.Integer, db.ForeignKey('permit.permit_id'), nullable=False)
+    received_date = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
+    issue_date = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
+    authorization_end_date = db.Column(db.DateTime, nullable=False, server_default=FetchedValue())
+    permit_amendment_status_code = db.Column(
+        db.String(2), db.ForeignKey('permit_amendment_status_code.permit_amendment_status_code'))
+    permit_amendment_type_code = db.Column(
+        db.String(2), db.ForeignKey('permit_amendment_type_code.permit_amendment_type_code'))
+
+    def json(self):
+        return {
+            'permit_amendment_id':
+            str(self.permit_amendment_id),
+            'permit_amendment_guid':
+            str(self.permit_amendment_guid),
+            'permit_id':
+            self.permit_id,
```

Comment:
Shouldn't this be wrapped in str() also?

Reply:
Is there a reason why we'd want the sequential IDs to be converted from integers to strings?

Reply:
Consistency since the first id is converted to a string.

Reply:
My question applies to both IDs in that case ;) I'd expect IDs that are ints in the backend to be returned as JavaScript Numbers in the JSON payload, personally. So, I'd like to make sure we're making an intentional choice to do this.

If it's a matter of "that's how we've done it elsewhere" then I would suggest we ask ourself if that's how we want to do it before it's been done in too many places.

In `frontend/src/customPropTypes/permits.js` `permit_id` is a number type. So, I get the impression that this is Ryan's intentional behaviour for this ID. But I'll leave it to him to weigh in now.

Reply:
I don't see why we would want to represent it as a string. str(self.permit_amendment_id) was copy/paste without thought and I will change it. IMO our API should properly return the few types JSON defines such as numbers, booleans, and null types, not numbers as strings, the word "True", or empty strings, etc.